admin/controllers/school.js

```diff
@@ -449,14 +450,55 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  const pupilsData = await fetchMultiplePupils(Object.values(req.body.pupil))
+  let pupilsToSave = []
+
+  await pupilsData.map(async (pupil) => {
+    if (pupil) {
+      pupil.attendanceCode = {
+        _id: req.body.attendanceCode,
+        dateRecorded: new Date(todayDate),
+        byUser: req.user.UserName
+      }
+      await pupilsToSave.push(pupil)
+    }
+  })
+
+  await Pupil.create(pupilsToSave, function (err, pupil) {
```

Comment:
This looks weird - async _and_ a callback.  In any case it seems `Model.create([], cb)` is not a bulk update, mongoose will turn this into a loop and save each object individually.

Reply:
[model].create works the same as [model].insertMany, the second is more recommended for bulk inserts except that it won't deal with updates. This is the best method I found for a bulk update, please show me an example of a better one and I'll refactor.

From Mongoose website:
http://mongoosejs.com/docs/api.html#model_Model.create
http://mongoosejs.com/docs/api.html#model_Model.insertMany